tests/providers/test_ssn.py

```diff
@@ -113,6 +113,85 @@ def test_ssn(self):
             assert 1 <= int(serial) <= 9999
             assert area != '666'
 
+    def test_invalid_ssn(self):
+        self.factory.seed(19030)
+        for _ in range(100):
```

Comment:
after setting `self.factory.seed()`, `self.factory.ssn` should always return the same value. I don't think this loop is necessary.

Reply:
I can set the seed() to 2432 to return an SSN that begins with 666, but removing the loop then tests only one SSN, and many asserts are not processed. Removing the seed(), and keeping the loop allows for 100 random Invalid SSNs to be tested, similar to the "test_ssn" unittest. Is that an acceptable approach?

Reply:
I'd rather set the seed multiple times for the specific assertions. Something like:

```python
self.factory.seed_instance(242)
ssn = self.factory.ssn(taxpayer_identification_number_type='INVALID_SSN')

assert ssn.startswith('666')

self.factory.seed(19030)
ssn = self.factory.ssn(taxpayer_identification_number_type='INVALID_SSN')
# assert something else
```

Reply:
> I'd rather set the seed multiple times for the specific assertions. Something like:
> 
> ```python
> self.factory.seed_instance(242)
> ssn = self.factory.ssn(taxpayer_identification_number_type='INVALID_SSN')
> 
> assert ssn.startswith('666')
> 
> self.factory.seed_instance(19030)
> ssn = self.factory.ssn(taxpayer_identification_number_type='INVALID_SSN')
> # assert something else
> ```

I believe I have corrected the assertions as shown above:

[New Commit](https://github.com/darrylwhiting/faker/commit/3dc30c69da300f90b6bbb4db2eacd06c8f771733)